vuepress/docs/docs/schema.md

```diff
@@ -164,6 +139,55 @@ An interface will be created with all the fields that are common to the case cla
 
 If you prefer to have a `Union` type instead of an `Enum`, even when the sealed trait contains only objects, add the `@GQLUnion` annotation.
 
+## Case classes and sealed traits
+
+The transformation between Scala types and GraphQL types is handled by a typeclass named `Schema`. As mentioned earlier, caliban provides instances of `Schema` for all basic Scala types, but inevitably you will need to support your own types, in particular case classes and sealed traits.
+
+Caliban is able to generate instances of `Schema` for case classes and sealed traits. You have 2 choices for doing that: auto derivation and semi-auto derivation.
+
+### Auto derivation
+Auto derivation is achieved easily by adding the following import:
+```scala mdoc:silent
+import caliban.schema.Schema.auto._
+```
+Using this import, caliban will generate a `Schema` instance for all the case classes and sealed traits that are found inside your resolver.
+
+::: warning Auto derivation limitations
+Auto derivation is the easiest way to get started, but it has some drawbacks:
+- If a type is referenced in several places inside your resolver, a `Schema` will be generated for each occurrence, which can lead to longer compilation times and a high amount of generated code (a sign of this is that the compiler will suggest increasing `-Xmax-inlines` in Scala 3).
+- When a `Schema` is missing for a nested type inside your resolver, it can sometimes be difficult to find out which type is missing when using auto derivation, because the error message will mention the root type and not the nested one.
+- The macro that generates the `Schema` instances sometimes gets confused when there are a lot of nested or recursive types, and can mistakenly generate a `Schema` for types that already have a `Schema` in scope. For this reason, semi-auto derivation is recommended for non-trivial schemas.
+:::
+
+### Semi-auto derivation
+Semi-auto derivation is achieved as follows for each type that needs a `Schema` instance (`MyClass` in the example):
+
+<code-group>
+  <code-block title="Scala 2" active>
+
+```scala mdoc:silent
+import caliban.schema.Schema
+
+case class MyClass(field: String)
+
+implicit val schemaForMyClass: Schema[Any, MyClass] = Schema.gen
+```
+  </code-block>
+  <code-block title="Scala 3">
+
+```scala
+import caliban.schema.Schema
+
+case class MyClass(field: String) derives Schema.SemiAuto
```

Comment:
Should we also mention `derives Schema.Auto`? I found it particularly useful for enums where it's cumbersome to create a custom schema for each individual enum value